File: cpp_version/floyd_warshall.cpp
#include <iostream>

using namespace std;

class Graph
{
private:
    const int INF = 9999999;
    const int V = 4;
    int graph[4][4] =
    {
        {0, 3, INF, 5},
        {2, 0, INF, 4},
        {INF, 1, 0, INF},
        {INF, INF, 2, 0}
    };

public:
    void PrintGraph()
    {
        for (int i = 0; i < V; ++i)
        {
            for (int j = 0; j < V; ++j)
            {
                if (graph[i][j] == INF)
                {
                    cout << "INF" << " ";
                }
                else
                {
                    cout << graph[i][j] << " ";
                }

            }
            cout << endl;

        }

    }
    void floyd_warshall_algo()
    {
        for (int k = 0; k < V; ++k)
        {
            for (int i = 0; i < V; ++i)
            {
                for (int j = 0; j < V; ++j)
                {
                    graph[i][j] = min(graph[i][k] + graph[k][j], graph[i][j]);

                }

            }

        }
        PrintGraph();

    }




};

int main(void)
{
    Graph g;
    g.floyd_warshall_algo();

    return 0;
}

File: python_version/floyd_warshall.py
#!/usr/bin/env python3

class Graph:
    INF = 9999999

    def __init__(self):
        self.V = 4
        self.graph = [[0, 3, self.INF, 5],
                      [2, 0, self.INF, 4],
                      [self.INF, 1, 0, self.INF],
                      [self.INF, self.INF, 2, 0]]

    def floyd_warshall(self):
        for k in range(self.V):
            for i in range(self.V):
                for j in range(self.V):
                    self.graph[i][j] = min(
                        self.graph[i][k] + self.graph[k][j], self.graph[i][j])
        for i in range(self.V):
            for j in range(self.V):
                if self.INF == self.graph[i][j]:
                    print("INF", end=' ')
                else:
                    print(self.graph[i][j], end=' ')
            print()


g = Graph()
g.floyd_warshall()
